src/__tests__/test.ts

```diff
@@ -92,6 +92,30 @@ describe("sanitizeUrl", () => {
     );
   });
 
+  it("decodes html entities", () => {
+    // all these decode to javascript:alert('xss');
+    const attackVectors = [
+      "&#0000106&#0000097&#0000118&#0000097&#0000115&#0000099&#0000114&#0000105&#0000112&#0000116&#0000058&#0000097&#0000108&#0000101&#0000114&#0000116&#0000040&#0000039&#0000088&#0000083&#0000083&#0000039&#0000041",
+      "&#106;&#97;&#118;&#97;&#115;&#99;&#114;&#105;&#112;&#116;&#58;&#97;&#108;&#101;&#114;&#116;&#40;&#39;&#88;&#83;&#83;&#39;&#41;",
+      "&#x6A&#x61&#x76&#x61&#x73&#x63&#x72&#x69&#x70&#x74&#x3A&#x61&#x6C&#x65&#x72&#x74&#x28&#x27&#x58&#x53&#x53&#x27&#x29",
+      "jav&#x09;ascript:alert('XSS');",
+      " &#14; javascript:alert('XSS');",
+    ];
```

Comment:
qq: Is there a reference to how these attack vector urls are generated?

Reply:
They are HTML encoded characters. So each character can be rendered as itself or as an HTML entity. For instance, `j` can be `&#106;` or `&#x6A;` or `&#0000106` (see https://www.htmlsymbols.xyz/unicode/U+006A)

Reply:
I wish I had an easy go-to site for generating them, but since `j` is a valid encoded character, by default, encoders don't convert it. Had trouble finding a place that would encode it using the special characters.

Most of these values came from the XSS report, and they do decode to the expected values.